src/asmcnc/comms/serial_connection.py

```diff
@@ -845,6 +855,49 @@ def process_grbl_push(self, message):
                     self.feed_rate = feed_speed[0]
                     self.spindle_speed = feed_speed[1]
 
+                # TEMPERATURES
+                elif part.startswith('TC:'):
+                    temps = part[3:].split(',')
+
+                    try: 
+                        float(temps[0])
+                        float(temps[1])
+                    except:
+                        log("ERROR status parse: Temperature invalid: " + message)
+                        return
+
+                    self.pcb_temp = float(temps[0])
+                    self.motor_driver_temp = float(temps[1])
+
+                    try:
+                        float(temps[2])
+                        self.transistor_heatsink_temp = float(temps[2])
+
+                    except IndexError:
+                        pass
```

Comment:
intentionally passing?

Reply:
(not certain on context but may be worth logging regardless)

Reply:
yes - so in the FW versions 1.3.6 - now, they've only reported 2 temps instead of 3, so this will try to retrieve the 3rd temp but if it isn't there then it'll just keep the code going :) 

I'm fairly sure this is the most "pythonic" way to handle this case

Reply:
Yep that makes sense, for completeness sake is there any reason why it would help to see a log of this reason? For example if a machine hasn't been updated?

Reply:
Nah, would just be extra noise at the mo - not all FW can go on all PCBs anyway, and currently we don't update FW in the field